setup.py

```diff
@@ -22,6 +22,10 @@ def read(filename):
       url=URL,
       download_url=URL + "/tarball/" + __version__,
       author=", ".join(MAIN_AUTHORS),
+      install_requires=["cryptography>=1.5",
```

Comment:
Just my 5 cents:
IMHO this would create a pain of keeping the `requirements.txt` and `setup.py` in-sync. Or in other words if one gets updated the other also needs to be updated which is undesirable.
instead perhaps read the requirements outside the setup-function and pass them in. Example:
```python
requirements = []
with open('requirements.txt') as f:
  requirements = f.read().splitlines()


setup(
    ...
    install_requires=requirements,
    ...
)
```

Reply:
Though I do agree that they could easily fall out of sync, there is, so far as I understand it, a difference between the two. See: [Vincent's comment](https://github.com/pypa/pipenv/issues/209#issuecomment-337409290), [Dan's comment](https://github.com/pypa/pipenv/issues/1263#issuecomment-362843247) or [this article](https://caremad.io/posts/2013/07/setup-vs-requirement/).

Considering your comment, and having dug a bit deeper, I would propose something like the requirements.txt-equivalent of the Pipfile in [this](https://github.com/pypa/pipenv/issues/209#issuecomment-337409290) comment.